File: dymo-api/nodejs-sdk/basic-signup-verification.ts
import DymoAPI from "dymo-api";

const dymo = new DymoAPI({
    apiKey: "PRIVATE_TOKEN_HERE"
});

async function checkSignUp({
    email,
    phone
}: {
    email: string;
    phone: string;
}): Promise<{
    pass: boolean;
    message?: string;
    realEmail?: string;
}> {
    try {
        const response = await dymo.isValidData({
            email,  // User's email address.
            phone   // If requested by the user (recommended).
        });

        // Email checks.
        if (!response.email.valid) return { pass: false, message: "Use a valid email." };
        if (response.email.fraud) return { pass: false, message: "Use your real email." };
        if (response.email.proxiedEmail) return { pass: false, message: "Use your real email." };

        // Phone checks.
        if (!response.phone.valid) return { pass: false, message: "Phone is not valid." };
        if (response.phone.fraud) return { pass: false, message: "Use your real phone." };

        return { pass: true, realEmail: response.email.email };
    } catch (error) {
        console.error(error);
        return { pass: true };
    }
};

File: dymo-api/python-sdk/basic-signup-verification.py
from dymoapi import DymoAPI

dymo = DymoAPI({
    "api_key": "PRIVATE_TOKEN_HERE"
})

def check_sign_up(email: str, phone: str) -> dict:
    try:
        response = dymo.is_valid_data({
            "email": email,  # User's email address.
            "phone": phone   # If requested by the user (recommended).
        })

        # Email checks.
        if not response.email.valid: return { "pass": False, "message": "Use a valid email." }
        if response.email.fraud: return { "pass": False, "message": "Use your real email." }
        if response.email.proxiedEmail: return { "pass": False, "message": "Use your real email." }

        # Phone checks.
        if not response.phone.valid: return { "pass": False, "message": "Phone is not valid." }
        if response.phone.fraud: return { "pass": False, "message": "Use your real phone." }

        return { "pass": True, "realEmail": response.email.email }

    except Exception as e:
        print(f"Error during signup check: {e}")
        return { "pass": True }